SEED-platform/seed
# #1882 - Added support for "" and !"" for date filters

#### What's this PR do?
- Added support for `""` and `!""` for date filters
- Added date-filtering support for `date` type columns (e.g. _Year Ending_), `datetime` type columns were already supported
- Added support for decimals in comparison (e.g. Latitude `>= 41.76`)
- Improved date support involving timezones (adjacent days could be equal if not handled properly)
- Added comprehensive tests

[Generic column syntax](https://regexr.com/3j1tq)
[Date/time column syntax:](https://regexr.com/50ok6)

#### How should this be manually tested?
1. Import data with dates
2. Use the column filters on the inventory list view to ensure that `""` and `!""` properly show empty/non-empty rows
3. Monkey test other filter cases
#### What are the relevant tickets?
#1882
